src/test/groovy/unit/org/pih/warehouse/order/OrderSpec.groovy

```diff
@@ -0,0 +1,88 @@
+package unit.org.pih.warehouse.order
+
+import grails.testing.gorm.DomainUnitTest
+import org.pih.warehouse.order.Order
+import org.pih.warehouse.order.OrderItem
+import org.pih.warehouse.order.OrderItemStatusCode
+import spock.lang.Specification
+
+class OrderSpec extends Specification implements DomainUnitTest<Order> {
+    void "Order.getCanGenerateInvoice() should return true when there is an order adjustment that hasn't already been invoiced"() {
+        given:
+        Order order = Spy(Order) {
+            getHasPrepaymentInvoice() >> hasPrepaymentInvoice
+        }
+        Set<OrderItem> orderItems = [
+                Mock(OrderItem),
+                Stub(OrderItem) {
+                    getHasNotInvoicedAdjustment() >> hasNotInvoicedAdjustment
+                },
+                Mock(OrderItem)
+        ]
+        order.setOrderItems(orderItems)
+
+        expect:
+        order.canGenerateInvoice == canGenerateInvoice
+
+        where:
+        hasPrepaymentInvoice | hasNotInvoicedAdjustment || canGenerateInvoice
+        false                | true                     || false
+        true                 | false                    || false
+        true                 | true                     || true
+    }
+
+    void "Order.getCanGenerateInvoice() should return true when there is not invoiced order item"() {
+        given:
+        Order order = Spy(Order) {
+            getHasPrepaymentInvoice() >> hasPrepaymentInvoice
+        }
+        OrderItem orderItem = Spy(OrderItem) {
+            getIsEncumbered() >> isEncumbered
+            getAllInvoiceItems() >> []
+        }
+        orderItem.orderItemStatusCode = orderItemStatusCode
+        Set<OrderItem> orderItems = [
+                Mock(OrderItem),
+                orderItem,
+                Mock(OrderItem)
+        ]
+        order.setOrderItems(orderItems)
+
+        expect:
+        order.canGenerateInvoice == canGenerateInvoice
+
+        where:
+        hasPrepaymentInvoice | orderItemStatusCode          | isEncumbered || canGenerateInvoice
+        false                | OrderItemStatusCode.CANCELED | false        || false
+        true                 | OrderItemStatusCode.CANCELED | false        || false
+        true                 | OrderItemStatusCode.PENDING  | true         || false
```

Comment:
what about testing the scenario where `Order.stats == PLACED`? then you would be able to generate an invoice right?

Reply:
Order status cannot be placed. The available options are:
```
[PENDING, COMPLETED, CANCELED, BACKORDER]
```

Reply:
that's OrderItemStatusCode right? I meant OrderStatus